Tests/LibC/TestScanf.cpp

```diff
@@ -26,7 +26,8 @@ constexpr static Array<unsigned char, 32> to_value_t(T x)
         T t;
     };
 
-    auto value = Value { .t = x };
+    Value value { .v = { 0 } };
+    value.t = x;
```

Comment:
... What's going on here. Isn't this plain UB, as we're reading from the inactive `v` member of the union below?

Reply:
Not entirely sure, tell me if I should change this in any way
(This whole file is a bit cursed imo)

Reply:
This is just type punning, isn't it? `constexpr` on this function is bogus but this should be fine in runtime

Reply:
> This is just type punning, isn't it? constexpr on this function is bogus but this should be fine in runtime

True, GCC does accept union-based type punning even if it's actually forbidden by _standard_ strict aliasing rules. But this pattern is weirder than that, as the char buffer is larger than the element type...

Reply:
The whole test file looks suspect to me, it might be in need of a refactor...

Reply:
Should I change something here or is it fine for now?